src/k-provider/ovp/provider.js

```diff
@@ -74,16 +74,22 @@ export default class OVPProvider extends BaseProvider<OVPProviderMediaInfoObject
     });
   }
 
-  doRequest(loaders: Array<RequestLoader>): Promise<any> {
-    const dataLoader = new OVPDataLoaderManager(this.playerVersion, this.partnerId, this.ks, this._networkRetryConfig);
+  doRequest(loaders: Array<RequestLoader>, ks?: string): Promise<any> {
+    let theKs: string = ks || this.ks;
+    const dataLoader = new OVPDataLoaderManager(this.playerVersion, this.partnerId, theKs, this._networkRetryConfig);
 
     return new Promise((resolve, reject) => {
+      if (!theKs) {
+        dataLoader.add(OVPSessionLoader, {widgetId: this.widgetId});
+      }
       loaders.forEach((loaderRequest: RequestLoader) => {
+        loaderRequest.params.ks = this.ks || '{1:result:ks}';
```

Comment:
```suggestion
        loaderRequest.params.ks = theKs || '{1:result:ks}';
```
add this validation to the tests